python/k8s.py

```diff
@@ -202,69 +221,73 @@ def install_docker_registry_secret(registry_provider, username, password):
-        '--k3s-agent-arg', '--kubelet-arg=eviction-minimum-reclaim=imagefs.available=1%,nodefs.available=1%',
-    ])  # yapf: disable
-    traefik_conf = ''
-    time.sleep(10)
-    while not traefik_conf:
-        try:
-            traefik_conf = config.kubectl.output(['get', 'cm', 'traefik', '-n', 'kube-system', '-o', 'yaml'])
-        except subprocess.CalledProcessError:
-            time.sleep(5)
-    traefik_conf = yaml.load(traefik_conf, Loader=yaml.FullLoader)
-    traefik_conf['data']['traefik.toml'] = ('insecureSkipVerify = true\n' + traefik_conf['data']['traefik.toml'])
-    with temporary_file() as f:
-        f.write(yaml.dump(traefik_conf).encode('utf8'))
-        f.close()
-        config.kubectl.call(['apply', '-n', 'kube-system', '-f', f.name])
-    config.kubectl.call(['delete', 'pod', '-l', 'app=traefik', '-n', 'kube-system'])
-
+    if config.k8s.distribution == "k3d":
+        """Create a k3d cluster"""
+        if name in [cluster['name'] for cluster in json.loads(check_output(split('k3d cluster list -o json')))]:
+            if recreate:
+                call(["k3d", "cluster", "delete", name])
+            else:
+                LOGGER.info(f"A cluster with the name {name} already exists. Nothing to do.")
+                return
+
+        if not is_port_available(80):
+            raise click.ClickException("Port 80 is already in use by another process. Please stop this process and retry.")
+        if not is_port_available(443):
+            raise click.ClickException("Port 443 is already in use by another process. Please stop this process and retry.")
+
+        import yaml
+        call([
+            'k3d', 'cluster', 'create', name,
+            '--wait',
+            '--port', '80:80@loadbalancer',
+            '--port', '443:443@loadbalancer',
+            '--registry-use', 'k3d-registry.localhost:5000',
+            '--k3s-agent-arg', '--kubelet-arg=eviction-hard=imagefs.available<1%,nodefs.available<1%',
+            '--k3s-agent-arg', '--kubelet-arg=eviction-minimum-reclaim=imagefs.available=1%,nodefs.available=1%',
+        ])  # yapf: disable
+        traefik_conf = ''
+        time.sleep(10)
+        while not traefik_conf:
+            try:
+                traefik_conf = config.kubectl.output(['get', 'cm', 'traefik', '-n', 'kube-system', '-o', 'yaml'])
+            except subprocess.CalledProcessError:
+                time.sleep(5)
+        traefik_conf = yaml.load(traefik_conf, Loader=yaml.FullLoader)
+        traefik_conf['data']['traefik.toml'] = ('insecureSkipVerify = true\n' + traefik_conf['data']['traefik.toml'])
+        with temporary_file() as f:
+            f.write(yaml.dump(traefik_conf).encode('utf8'))
+            f.close()
+            config.kubectl.call(['apply', '-n', 'kube-system', '-f', f.name])
+        config.kubectl.call(['delete', 'pod', '-l', 'app=traefik', '-n', 'kube-system'])
+    else:
+        LOGGER.error("Unsupported distribution")
+        return
```

Comment:
use `raise click.ClickException("Unsupported distribution")`  instead of these two lines

Reply:
Ok, fixed :)